earth_enterprise/src/fusion/autoingest/sysman/AssetD.cpp

```diff
@@ -15,23 +15,22 @@
 
 #include "AssetD.h"
 #include <AssetThrowPolicy.h>
-
+#include <memory>
 // ****************************************************************************
 // ***  AssetImplD
 // ****************************************************************************
 
-khRefGuard<AssetImplD>
+std::shared_ptr<AssetImplD>
 AssetImplD::Load(const std::string &boundref)
 {
-  khRefGuard<AssetImplD> result;
-
   // make sure the base class loader actually instantiated one of me
   // this should always happen, but there are no compile time guarantees
-  result.dyncastassign(AssetImpl::Load(boundref));
+  std::shared_ptr<AssetImplD> result =
```

Comment:
I think the issue is here (and the other places where dyncastassign used to be called). I don't know why it's failing now, but apparently the dynamic_pointer_cast is returning nullptr.